service/src/main/kotlin/fi/espoo/evaka/backuppickup/BackupPickupController.kt.kt

```diff
@@ -0,0 +1,157 @@
+            tx.createBackupPickup(
+                body
+            )
+        }.let {
+            ResponseEntity.ok(ChildBackupPickupCreateResponse(it))
+        }
+    }
+
+    @PutMapping("/children/{childId}/backup-pickups")
+    fun updateForChild(
+        db: Database.Connection,
+        user: AuthenticatedUser,
+        @PathVariable("childId") childId: UUID,
+        @RequestBody body: ChildBackupPickup
+    ): ResponseEntity<Unit> {
+        Audit.ChildBackupPickupUpdate.log(targetId = childId)
+        acl.getRolesForChild(user, childId).requireOneOfRoles(UserRole.ADMIN, UserRole.STAFF, UserRole.SPECIAL_EDUCATION_TEACHER)
+        return db.transaction { tx ->
+            tx.updateBackupPickup(
+                body
+            )
+        }.let {
+            ResponseEntity.noContent().build()
+        }
+    }
+
+    @DeleteMapping("/children/{childId}/backup-pickups/{id}")
+    fun delete(
+        db: Database.Connection,
+        user: AuthenticatedUser,
+        @PathVariable("id") id: UUID,
+        @PathVariable("childId") childId: UUID
+    ): ResponseEntity<Unit> {
+        Audit.ChildBackupPickupDelete.log(targetId = childId)
+        acl.getRolesForChild(user, childId).requireOneOfRoles(UserRole.ADMIN, UserRole.STAFF, UserRole.SPECIAL_EDUCATION_TEACHER)
+        db.transaction { tx ->
+            tx.deleteBackupPickup(id)
+        }
+        return ResponseEntity.noContent().build()
+    }
+}
+
+fun Database.Transaction.deleteBackupPickup(
+    id: UUID
+) {
+    val deleted = this.createUpdate("DELETE FROM backup_pickup WHERE id = :id")
+        .bind("id", id)
+        .execute()
+
+    if (deleted == 0) throw NotFound("No backup pickup $id found")
+}
+
+fun Database.Read.getBackupPickupsForChild(childId: UUID): List<ChildBackupPickup> {
+    return createQuery("SELECT * FROM backup_pickup WHERE child_Id = :id")
+        .bind("id", childId)
+        .mapTo<ChildBackupPickup>()
+        .list()
+}
+
+fun Database.Transaction.createBackupPickup(
+    backupPickup: NewChildBackupPickup
+): UUID {
+    // language=sql
+    val sql = """
+        INSERT INTO backup_pickup (child_id, name, phone)
+        VALUES (:childId, :name, :phone)
+        RETURNING id
+    """.trimIndent()
+    return this.createQuery(sql)
+        .bind("childId", backupPickup.childId)
+        .bind("name", backupPickup.name)
+        .bind("phone", backupPickup.phone)
+        .mapTo<UUID>()
+        .first()
+}
+
+fun Database.Transaction.updateBackupPickup(
+    backupPickup: ChildBackupPickup
+): Int {
+    // language=sql
+    val sql = """
+        UPDATE backup_pickup
+        SET
+            child_id = :childId,
+            name = :name,
+            phone = :phone
+        WHERE id  = :id
+    """.trimIndent()
+    return this.createUpdate(sql)
+        .bind("id", backupPickup.id)
+        .bind("childId", backupPickup.childId)
+        .bind("name", backupPickup.name)
+        .bind("phone", backupPickup.phone)
+        .execute()
```

Comment:
this returns Int but the value is not checked in controller. Could also already throw NotFound here as you did with delete